zetaclient/chains/bitcoin/observer/inbound.go

```diff
@@ -476,13 +486,56 @@ func GetBtcEvent(
 		}
 
 		return &BTCInboundEvent{
-			FromAddress: fromAddress,
-			ToAddress:   tssAddress,
-			Value:       value,
-			MemoBytes:   memo,
-			BlockNumber: blockNumber,
-			TxHash:      tx.Txid,
+			FromAddress:  fromAddress,
+			ToAddress:    tssAddress,
+			Value:        value,
+			DepositorFee: depositorFee,
+			MemoBytes:    memo,
+			BlockNumber:  blockNumber,
+			TxHash:       tx.Txid,
 		}, nil
 	}
 	return nil, nil
 }
+
+// GetSenderAddressByVin get the sender address from the previous transaction
+func GetSenderAddressByVin(rpcClient interfaces.BTCRPCClient, vin btcjson.Vin, net *chaincfg.Params) (string, error) {
```

Comment:
The method `GetSenderAddressByVin` was moved from `observer.go` to `inbound.go` as it's intended for inbound observation.